packages/@remirror/core/src/builtins/__tests__/persistent-selection-extension.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { extensionValidityTest, renderEditor } from 'jest-remirror';
+
+import { PersistentSelectionExtension } from '..';
+
+extensionValidityTest(PersistentSelectionExtension);
+
+test('adds decoration when configured', () => {
+  const builtin = { persistentSelectionClass: 'persistent-selection' };
+  const {
+    add,
+    selectText,
+    view: { props },
+    state,
+    nodes: { p, doc },
+  } = renderEditor([], { builtin });
+
+  add(doc(p('Some text')));
+
+  selectText(4);
+
+  setTimeout(() => {
```

Comment:
Not sure that we need the `setTimeout` here. Updates are immediately available. 

Also, the test seems to be focused on the implementation details. IMO the important things to test are:

- The current selection is wrapped with an element that has the class `persistent-selection`. So `expect(view.dom.innerHTML).toMatchInlineSnapshot()` is a good enough test.
- When focus is lost the selection class is maintained. I think this was the main goal of the original issue.

Reply:
I had to add the timeout, neither `act()` nor simply chaining the calls passed the tests.

Reply:
> When focus is lost the selection class is maintained.

Added a test for this case as well now.